doc/reference/make.md

```diff
@@ -9,36 +9,184 @@ See also the [guide](/guide/build.md).
+    before any of the submodule targets are compiled.
+
+  - A list of the form `(exe: file opts ...)` where the file is a Gerbil Scheme module source file
+    (with the `".ss"` extension automatically added by default) that exports a function `main`.
+    The file will be compiled by `gxc` with the provided options into an executable, that,
+    when invoked, calls the `main` function with a single argument, the list of command-line parameters
+    (excluding the `argv[0]` command location). Note that this executable will dynamically load
+    libraries from your `GERBIL_PATH` and/or your `~/.gerbil/lib` as needed.
+
+  - A list of the form `(static-exe: file opts ...)` which is similar to the `exe:` case above,
+    except that the executable will statically include all the compiled Gerbil code rather than
+    dynamically load it from your Gerbil path, etc.
+    Note that from the point of view of the "C" world, it is still a "dynamically linked" executable
+    that is linked against the `libc`, `libgambit`, etc. ---
+    only the Scheme code is statically compiled.
+
+  - A list of the form `(copy: file)` which specifies a file that will be copied
+    to the Gerbil library path. It can be a precompiled C object, Scheme file, data file, etc.,
+    that your library uses and needs be present wherever your library is installed.
+
+### Build Settings
+
+The `make` function and the `defbuild-script` macro after it accept
+the following keyword arguments, that may configure how your project is built.
+
+  - `srcdir:` specifies the top directory under which to find the source code
+    for your project, such that if you specify a target `"foo/bar"` or `"foo/bar.ss"`
+    then it will refer to a file `"foo/bar.ss"` under this directory.
+    This argument is mandatory when calling `make`, and is the only mandatory settings argument.
+    However, this argument is prohibited when using the `defbuild-script` macro,
+    since the macro uses `(path-normalize (path-directory (this-source-file)))`
+    to automatically extract this directory from the current `./build.ss` file containing it
+    and pass it to `make` (and you cannot override it).
+    You may use the same trick in your `./build.ss` if for some reason you don't use that macro.
+    (The trick is similar to `$(realpath "$(dirname "${BASH_SOURCE[0]}")")` in *bash*
+    or `$(realpath "$(dirname "$0")")` in *zsh*).
+
+  - `prefix:` specifies a prefix to the path of your Gerbil modules.
+    This prefix will affect how your modules are named and where they will be installed
+    within your *libdir*. Thus, a prefix of `legs` means that a file `"foo/bar.ss"`
+    will be compiled into a module `legs/foo/bar` that you can import with `(import :legs/foo/bar)`
+    and that will be installed under your *libdir* as a bunch of files starting with the prefix
+    `legs/foo/bar`. The default, `#f`, means that no prefix will be prepended.
```

Comment:
`defbuild-script` should be automatically detecting that from gerbil.pkg; perhaps we can do the same for `make` at large and make this very much an edge case.

Reply:
Yes, `defbuild-script` does that and I will document it. Shall I modify `make` to do it, too? Getting the package from `gerbil.pkg` in the `srcdir` ? What if the `gerbil.pkg` is actually in a parent directory? Is that allowed?

Reply:
Done, as a separate commit.